internal/server/destinations.go

```diff
@@ -79,7 +82,7 @@ func (a *API) UpdateDestination(c *gin.Context, r *api.UpdateDestinationRequest)
 	rCtx := getRequestContext(c)
 
 	// Start with the existing value, so that non-update fields are not set to zero.
-	destination, err := access.GetDestination(c, r.ID)
+	destination, err := data.GetDestination(rCtx.DBTxn, data.GetDestinationOptions{ByID: r.ID})
```

Comment:
This is an example of where the `access` package is not working well. Ideally we would authorize before doing any work, but we can see here that even though we were calling an `access` function first, the correct authorization does not happen until the call below here on line 98.

I think the API handlers should be authorizing first, either declaratively from something on the `route`, or imperatively as the first operation in the handler.